tests/mir-opt/const_prop/slice_len.main.GVN.32bit.panic-abort.diff

```diff
@@ -22,12 +22,11 @@
           StorageLive(_3);
           StorageLive(_4);
           _9 = const main::promoted[0];
--         _4 = _9;
+          _4 = _9;
 -         _3 = _4;
 -         _2 = move _3 as &[u32] (PointerCoercion(Unsize));
-+         _4 = const {ALLOC0<imm>: &[u32; 3]};
-+         _3 = const {ALLOC0<imm>: &[u32; 3]};
-+         _2 = const {ALLOC0<imm>: &[u32; 3]} as &[u32] (PointerCoercion(Unsize));
++         _3 = _9;
++         _2 = _9 as &[u32] (PointerCoercion(Unsize));
```

Comment:
@saethlin this is this kind of pattern.
The constant is still there, named `main::promoted[0]`. It's "evaluated" form `const {ALLOC0<imm>: &[u32; 3]}` does not appear in the MIR dump, so the dump does not print `ALLOC0`. This does not mean that `ALLOC0` was not evaluated, just that it's not put in MIR and not printed.

Reply:
So are you saying there consts that get their allocations dumped, and some consts that don't? That seems like an undesirable property of MIR dumps, right?

Reply:
yea, Ideally promoteds would just be regular statics, but lazily created in the MIR opt pipeline. This is blocked on various works around `DefId` creation, most importantly https://github.com/rust-lang/rust/pull/115613. I guess we could add a special case for showing promoteds like we show other allocations for now, but that seems orthogonal to this PR

Reply:
Statics can't be generic so IMO they should be regular consts.